src/workshed/index.ts

```diff
@@ -0,0 +1,164 @@
+    GarboWorkshed._nextWorkshed ??= GarboWorkshed.get(globalOptions.workshed);
+    return GarboWorkshed._nextWorkshed;
+  }
+
+  static useNext(): GarboWorkshed | null {
+    if (get("_workshedItemUsed")) return null;
+    const next = GarboWorkshed.next;
+    if (next && have(next.workshed)) {
+      use(next.workshed);
+      if (GarboWorkshed.get(getWorkshed()) === next) {
+        GarboWorkshed._nextWorkshed = null;
+        GarboWorkshed._currentWorkshed = next;
+      }
+    }
+    return GarboWorkshed._currentWorkshed;
+  }
+}
+const estimatedTurnsTomorrow = 400 + clamp((get("valueOfAdventure") - 4000) / 8, 0, 600);
+
+const worksheds = [
+  new GarboWorkshed({
+    workshed: $item`model train set`,
+    done: () => {
+      // We should always get value from the trainset, so we would never switch from it
+      return false;
+    },
+    action: () => {
+      if (!TrainSet.canConfigure()) return;
+      if (!get("trainsetConfiguration")) {
+        // Visit the workshed to make sure it's actually empty, instead of us having not yet seen it this run
+        visitUrl("campground.php?action=workshed");
+        visitUrl("main.php");
+      }
+      if (!get("trainsetConfiguration")) {
+        print("Reconfiguring trainset, as our next station is empty", "blue");
+        return TrainSet.setConfiguration(getBestCycle());
+      } else {
+        const bestTwoStations = getBestCycle().splice(0, 2);
+        const offset = get("trainsetPosition") % 8;
+        if (bestTwoStations.includes(TrainSet.next())) return;
+        print(`Reconfiguring trainset, as our next station is ${TrainSet.next()}`, "blue");
+        return TrainSet.setConfiguration(offsetDefaultPieces(offset));
+      }
+    },
+  }),
+  new GarboWorkshed({
+    workshed: $item`cold medicine cabinet`,
+    done: () => get("_coldMedicineConsults") >= 5,
+    action: () => {
+      if (get("_nextColdMedicineConsult") > totalTurnsPlayed()) return;
+      grabMedicine();
+    },
+  }),
+  new GarboWorkshed({
+    workshed: $item`Asdon Martin keyfob`,
+    done: () => {
+      return (
+        haveEffect($effect`Driving Observantly`) >=
+        estimatedTurns() + (globalOptions.ascend ? 0 : estimatedTurnsTomorrow)
+      );
+    },
+    action: () => {
+      AsdonMartin.drive(
+        $effect`Driving Observantly`,
+        estimatedTurns() + (globalOptions.ascend ? 0 : estimatedTurnsTomorrow)
+      );
+    },
+  }),
+  new GarboWorkshed({
+    workshed: $item`Little Geneticist DNA-Splicing Lab`,
+    done: () => {
+      // This will likely always return true or false for now, depending on the start state of garbo
+      // Since we don't actually support using the syringe in combat at this time, the counter will never change
+      return get("_dnaPotionsMade") >= 3;
+    },
+    action: () => {
+      // Just grab whatever tonics for now, since we don't actually have support for DNA
+      if (get("dnaSyringe")) DNALab.makeTonic(3);
+    },
+  }),
```

Comment:
I was also thinking of setting a global `let _triedMakingTonics = false`, with `done: () => { return _triedMakingTonics || get("_dnaPotionsMade") >=3; }` and `action: () => { _triedMakingTonics = true; if (get("dnaSyringe")) DNALab.makeTonic(3); }`, which should be much better behaviour that respects the user's intent to swap out of DNA

Reply:
I'm going to declare that out of scope and a good fit for future generations